neptune/internal/collections/disk_queue.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright (c) 2020, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from typing import TypeVar, List, Callable, Optional
+
+from neptune.internal.collections.queue import Queue
+from neptune.internal.utils.json_file_splitter import JsonFileSplitter
+
+T = TypeVar('T')
+
+
+class DiskQueue(Queue[T]):
```

Comment:
Consider exposing two interfaces, one for writing and one for reading. No strong feelings about this one though.